packages/material-ui/src/IconButton/IconButton.js

```diff
@@ -103,7 +104,25 @@ IconButton.propTypes = {
   /**
    * The icon element.
    */
-  children: PropTypes.node,
+  children: chainPropTypes(PropTypes.node, props => {
+    const found = React.Children.toArray(props.children).some(child => {
+      return React.isValidElement(child) && child.props.onClick;
+    });
+
+    if (found) {
+      return new Error(
+        [
+          'Material-UI: you are providing an onClick event listener ' +
+            'to a child of a button element.',
+          'Firefox will never trigger the event.',
+          'You should move the onClick listener to the parent button element.',
+          'https://github.com/mui-org/material-ui/issues/13957',
+        ].join('\n'),
```

Comment:
I would include a random seed here in the test environment to be able to trigger repeated warnings like we did in https://github.com/mui-org/material-ui/blob/5773bd0b4714a43bc11ef4bf509207a5336fd112/packages/material-ui-utils/src/componentPropType.js#L38

Reply:
Great idea!